rclpy/rclpy/timer.py

```diff
@@ -37,52 +36,44 @@ def __init__(self, callback, callback_group, timer_period_ns, clock, *, context=
 
     @property
     def handle(self):
-        return self.__handle
+        return self.__timer
 
     def destroy(self):
-        self.handle.destroy()
+        self.__timer.destroy_when_not_in_use()
 
     @property
     def clock(self):
         return self._clock
 
     @property
     def timer_period_ns(self):
-        with self.handle as capsule:
-            val = _rclpy.rclpy_get_timer_period(capsule)
-        self._timer_period_ns = val
+        val = self.__timer.get_timer_period()
```

Comment:
Please keep the `context` manager so that the timer can't be destroyed while this method is called.

```python
with self.__timer:
    val = self.__timer.get_timer_period()
```

Reply:
Same comment for the rest of the functions in this file